Assets/Scripts/Managers/InstanceManager.cs

```diff
@@ -100,11 +109,20 @@ public static class InstanceManager {
         }
 
         public static GameObject DisplayFullscreen(string name) {
-            return Instantiate(name, Canvas.transform.Find("Player UI").Find("Fullscreen"));
+            if (FullScreen) {
+                return Instantiate(name, FullScreen);
+            } else {
+                return null;
+            }
+            
         }
-
+        
         public static GameObject DisplayWindow(string name) {
-            return Instantiate(name, Canvas.transform.Find("Player UI").Find("View Area").Find("Middle Section").Find("Windows Section"));
+            if (FullScreen) {
+                return Instantiate(name, WindowsSection);
+            } else {
+                return null;
+            }
```

Comment:
I didn't quite mean it like this. The reason I suggested the code bit I posted before was that it stores the Display Window in a variable. By default it's `null` and so the first time the method is called it would Instantiate the GameObject and store the reference for it in a field. Later, since the GameObject is stored in the variable I won't have to be re-instantiated, you can just return the already stored GameObject (thus increasing performace). However, my other question asking whether this GameObject is then destroyed anywhere is important because then this method is a bit more tricky and the destroy part would have to be refactored to only disable the GameObject instead. Let me know if you need any clarifications

Reply:
You can't do that on a static class, if you referenced the window class then went to the main menu, you would then be referencing a destroyed object, which would cause errors. This way the window gameobject is referenced when the playercanvus prefrab is present on sceneload and is set to null when that prefab isn't. Your method isn't actually quicker, it just stores it on first call, whereas the way I have done it, calls it on sceneload. Does that clear it up?

Reply:
Once destroyed though, the reference should be nullified thus the first case of the if statement would be true again. It's not meant to be quicker it's meant to avoid reinstantiating of the same object and then destroying it over and over

Reply:
I think we might have crossed wires. Is the object your talking about the "Windows Section" object or the Instantiated task object?